### ISSUE TITLE
Debug console loses ability to differentiate lines of output

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#18442]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: Version: 1.75.1 (Universal)
- OS Version: macOS 13.0.1 (22A400) (M1 Pro)

Steps to Reproduce:

It's not 100% clear to me what the definitive recreate is. Initially, I thought it was something to do with code that outputs multiline log entries, but while testing with Extensions disabled, I realized that observation was a red herring.

I created a Loom to show the issue:
https://www.loom.com/share/66777e9fdb774a4186db493d02091667

In short, the Debug console starts by identifying individual output lines as I would expect; then it clumps a bunch of lines together as if they are one; then it goes back to single lines.

The impact is that output filtering doesn't work properly, which makes some debugging tasks very difficult.
